src/main/asciidoc/java/net.adoc

```diff
@@ -797,4 +797,54 @@ NetServerOptions options = new NetServerOptions().
 NetServer server = vertx.createNetServer(options);
 ----
 
-Cipher suite can be specified on the `link:../../apidocs/io/vertx/core/net/NetServerOptions.html[NetServerOptions]` or `link:../../apidocs/io/vertx/core/net/NetClientOptions.html[NetClientOptions]` configuration.
\ No newline at end of file
+Cipher suite can be specified on the `link:../../apidocs/io/vertx/core/net/NetServerOptions.html[NetServerOptions]` or `link:../../apidocs/io/vertx/core/net/NetClientOptions.html[NetClientOptions]` configuration.
+
+==== OpenSSL engine
+
+The default SSL/TLS engine implementation is provided by the JDK.
+
+The engine implementation can be configured to use https://www.openssl.org[OpenSSL] instead. OpenSSL provides
+better performances and CPU usage than the JDK engine, as well as JDK version independence.
+
+OpenSSL requires to configure `link:../../apidocs/io/vertx/core/net/TCPSSLOptions.html#setSslEngine-io.vertx.core.net.SSLEngine-[setSslEngine]` to `link:../../apidocs/io/vertx/core/net/SSLEngine.html#OPENSSL[OPENSSL]`
+and use http://netty.io/wiki/forked-tomcat-native.html[netty-tcnative] jar on the classpath, as well as the correct
+version of OpenSSL installed on your OS.
+
+OpenSSL restricts the key/certificate configuration to `.pem` files. However it is still possible to use any trust
+configuration.
+
+==== Application-Layer Protocol Negotiation
+
+ALPN is a TLS extension for applicationl layer protocol negotitation. It is used by HTTP/2: during the TLS handshake
+the client gives the list of application protocols it accepts and the server responds with a protocol it supports.
+
+Java 8 does not supports ALPN out of the box, so ALPN should be enabled by other means:
+
+- _OpenSSL_ support
+- _Jetty-ALPN_ support
+
+===== OpenSSL ALPN support
+
+OpenSSL provides native ALPN support.
+
+===== Jetty-ALPN support
+
+Jetty-ALPN is a small jar that overrides a few classes of Java 8 distribution to support ALPN.
+
+The JVM must be started with the _alpn-boot-${version}.jar_ in its `bootclasspath`:
+
+----
+-Xbootclasspath/p:/path/to/alpn-boot${version}.jar
+----
+
```

Comment:
If I understand correctly, `lib/ext` would not be high enough in the classloader hierarchy. Maybe it needs to be mentioned.